src/zarr/core/array.py

```diff
@@ -554,7 +554,12 @@ async def _create_v3(
         attributes: dict[str, JSON] | None = None,
         exists_ok: bool = False,
     ) -> AsyncArray[ArrayV3Metadata]:
-        if not exists_ok:
+        if exists_ok:
+            if store_path.store.supports_deletes:
+                await store_path.delete_dir(recursive=True)
+            else:
+                await ensure_no_existing_node(store_path, zarr_format=3)
+        else:
```

Comment:
[not a blocker for this PR but just an idea triggered by this code]

deleting an entire array can take `O(num_chunks)` time, and I know this has been a pain point for for zarr users including myself. To alleviate this, we should expose two settings:
- one setting for array creation that avoids a recursive delete, and instead only creates / modifies metadata documents
- one setting for chunks that causes zarr to ignore existing invalid chunks and simply overwrite them.

together, these two settings would make creating re-creating arrays in-place very cheap, albeit more likely to contain garbage from old arrays / groups, so if we do this we should also create a routine for cleaning up that garbage.

Reply:
Happy to discuss this idea in a separate issue by my initial reaction is that this sounds like a very dangerous option!

> To alleviate this, we should expose two settings:

The big change in 3.0 is that the store is async. So we can optimize this by more aggressively utilizing concurrency.

Reply:
> Happy to discuss this idea in a separate issue by my initial reaction is that this sounds like a very dangerous option!

I will open an issue for the idea. I agree that it's opens up a "hard hat zone" in the API. 👷